repos/system_upgrade/el9toel10/actors/sssd/sssdfacts/libraries/sssdfacts.py

```diff
@@ -0,0 +1,53 @@
+import os
+import re
+
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.stdlib import api
+from leapp.models import KnownHostsProxyConfig
+
+
+def _does_file_contain_expression(file_path, expression):
+    try:
+        with open(file_path) as in_file:
+            for line in in_file:
+                if re.search(expression, line) is not None:
+                    return True
+        return False
+    except FileNotFoundError:
+        api.current_logger().warning(
+            'Found a file during a recursive walk, but we failed to open it for reading: {}'.format(file_path)
+        )
+        return False
+    except OSError as e:
+        raise StopActorExecutionError('Could not open file ' + file_path, details={'details': str(e)})
```

Comment:
Nitpick (not blocking, we can change it separately). Also it would be better to have information about the file just in details, so the error msg is consistent across various potential files. but as this is kind of corner case situation (root has usually the access) it does not matter so much.

```suggestion
        raise StopActorExecutionError('Could not open file {}'.format(file_path), details={'details': str(e)})
```